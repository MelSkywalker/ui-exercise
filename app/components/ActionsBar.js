import React, {useState } from 'react';
import { useDispatch } from 'react-redux';
import { DELETE_MESSAGE } from '../redux/reducers';
import ImportantIcon from './svg/ImportantIcon';
import TrashIcon from './svg/TrashIcon';

import './actionsBar.scss';

const ActionsBar = () => {
	const dispatch = useDispatch();

	const handleRemoveItems = (e) => {
		console.log('deleting...');
		dispatch({
			type: DELETE_MESSAGE
		})
	};
	return (
		<div className="actions-bar">
			<input type="checkbox" className="select-all" />
			<button>
				<ImportantIcon />
			</button>
			<button className="delete-button" onClick={handleRemoveItems}>
				<TrashIcon />
			</button>
		</div>
	)
};

export default ActionsBar;
