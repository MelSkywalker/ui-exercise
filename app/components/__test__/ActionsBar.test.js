import React from 'react';
import { shallow } from 'enzyme';

import ActionsBar from '../ActionsBar';

describe('components | ActionsBar', () => {
	it('it renders the component', () => {
		const component = shallow(<ActionsBar />);
		expect(component).toMatchSnapshot();
	});
});
