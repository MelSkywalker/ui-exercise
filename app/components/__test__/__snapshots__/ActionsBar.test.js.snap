// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | ActionsBar it renders the component 1`] = `
<div
  className="actions-bar"
>
  <input
    className="select-all"
    type="checkbox"
  />
  <ImportantIcon />
  <button>
    <TrashIcon />
  </button>
</div>
`;
